types/wlr_surface.c

```diff
@@ -169,10 +169,20 @@ static void surface_state_viewport_src_size(struct wlr_surface_state *state,
 
 static void surface_state_finalize(struct wlr_surface *surface,
 		struct wlr_surface_state *state) {
+	if (state->buffer_resource) {
```

Comment:
I think we could move this to the `if ((state->committed & WLR_SURFACE_STATE_BUFFER))` below. No need to update `state->buffer` if it's still the same as before.

Reply:
Done. I think I put it there when I was debugging and forgot to move it after.